### DISCUSSION TITLE
Is it possible to stop a job from going to the dead queue inside sidekiq_retries_exhausted?

### DISCUSSION THREAD

**Discussion by concept47** [sidekiq/sidekiq#5696] [General]
Probably a bit odd request, but was just wondering about this in general. 

What I'm trying to do is stop a job from being duplicated in the Dead queue if there already exists a job with the same exact arguments in there already.

**Comment by mperham** [comment]
Off-hand, I can't think of an easy or efficient way to do this.

**Comment by concept47** [comment]
This seemed to work

```
  sidekiq_retries_exhausted do |job, _exception|
    dupe = Sidekiq::DeadSet.new.find { |dead_job| dead_job.args == job['args'] && dead_job.klass == job['class'] }

    job['dead'] = false if dupe.present?
  end
```
